spec/index.bs

```diff
@@ -1113,6 +1101,39 @@ Given an {{HTMLScriptElement}} (|script|), this algorithm performs the following
 
     If the algorithm threw an error, rethrow the error.
 
+## Get Trusted Types-compliant attribute value ## {#validate-attribute-mutation}
+To <dfn abstract-op export>get Trusted Types-compliant attribute value</dfn> on {{Attr}} |attribute| with {{Element}} |element| and {{TrustedType}} or a string |newValue|, perform the following steps:
+
+1. Let |expectedType| be null.
+1. Find the row in the following table, where |element| is in the first column, and |attribute|'s <a for="Attr">namespace</a> and <a for="Attr">local name</a> match the values in the second and third column, respectively.
+   If a matching row is found, set |expectedType| to the value of the fourth column in that row.
+
+    <table>
+      <thead>
+        <tr><th>Element<th>Attribute namespace<th>Attribute local name<th>TrustedType<th>
+      <tbody>
+        <tr><td>{{HTMLIFrameElement}}<td>null<td>"srcdoc"<td>{{TrustedHTML}}
+        <tr><td>{{HTMLEmbedElement}}<td>null<td>"src"<td>{{TrustedScriptURL}}
+        <tr><td>{{HTMLScriptElement}}<td>null<td>"src"<td>{{TrustedScriptURL}}
+        <tr><td>{{HTMLObjectElement}}<td>null<td>"data"<td>{{TrustedScriptURL}}
+        <tr><td>{{HTMLObjectElement}}<td>null<td>"codebase"<td>{{TrustedScriptURL}}
+        <tr><td>{{SVGScriptElement}}<td><a>XLink namespace</a><td>"href"<td>{{TrustedScriptURL}}
+      </tbody>
+    </table>
+
+1. If |expectedType| is null, then:
+    1. If |newValue| is a string, return |newValue|.
+    1. <a>Assert</a>: |newValue| is {{TrustedHTML}} or {{TrustedScript}} or {{TrustedScriptURL}}.
+    1. Return |value|'s `[[Data]]` internal slot value.
+1. Return the result of executing [$Get Trusted Type compliant string$] with the following arguments:
+     * |expectedType|
+     * |newValue| as |input|
+     * |element|'s <a>node document</a>'s <a>relevant global object</a> as |global|
+     * the result of [=concatenating=] « |element|'s [=Element/local name=], |attribute|'s [=Attr/local name=] » with `"."` as a |separator| as |sink|
```

Comment:
Yes, we're lucky. In general the sink name is just some approximation of where the violation is occurring. Since it's present in the violation report samples, which are trimmed, we wanted to come up with a short, close-enough identifier. We can add an assert here for SVG or null namespace of the attribute, nothing else comes to mind.

Reply:
Both good and bad news with the newest commit.

I've added sink name to the table and in fact, following https://github.com/w3c/trusted-types/issues/260#issuecomment-594737034m we were already using a longer form of the sink name in the implementation.

I've also verified that a `svg:script` executes code with `href` attribute from both null and xlink namespaces, so we do have a conflict in sink name already. Like explained before, I don't think it's a practical problem, because I can't find a reason why authors would treat (e.g. sanitize) both attributes values differently, branching on namespace.